codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoderTest.java

```diff
@@ -189,6 +194,24 @@ public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
                 return future;
             }
         });
+        when(writer.writeHeaders(eq(ctx), anyInt(), any(Http2Headers.class), anyInt(), anyShort(), anyBoolean(),
+                anyInt(), anyBoolean(), any(ChannelPromise.class)))
+                .then(new Answer<Object>() {
+                    @Override
+                    public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                        ChannelPromise receivedPromise = (ChannelPromise) invocationOnMock.getArguments()[8];
+                        if (streamClosed) {
+                            fail("Stream already closed");
+                        } else {
```

Comment:
@nmittler I basically just copied the code from `writeData`. Should I change it there as well or does it serve a specific purpose there?